h2o-core/src/main/java/water/api/schemas3/FrameV3.java

```diff
@@ -207,31 +207,39 @@ public void clearBinsField() {
   public FrameV3(Key<Frame> frame_id) { this.frame_id = new FrameKeyV3(frame_id); }
 
   public FrameV3(Frame fr) {
-    this(fr, 1, (int) fr.numRows(), 0, 0); // NOTE: possible row len truncation
+    this(fr, 1, (int) fr.numRows(), 0, -1, true); // NOTE: possible row len truncation
   }
 
   public FrameV3(Frame f, long row_offset, int row_count) {
-    this(f, row_offset, row_count, 0, 0);
+    this(f, row_offset, row_count, 0, -1, true);
   }
 
   public FrameV3(Frame f, long row_offset, int row_count, int column_offset, int column_count) {
-    this.fillFromImpl(f, row_offset, row_count, column_offset, column_count);
+    this.fillFromImpl(f, row_offset, row_count, column_offset, column_count, true);
+  }
+
+  public FrameV3(Frame f, long row_offset, int row_count, int column_offset, int column_count, boolean expensive) {
+    this.fillFromImpl(f, row_offset, row_count, column_offset, column_count, expensive);
   }
 
   @Override public FrameV3 fillFromImpl(Frame f) {
-    return fillFromImpl(f, 1, (int)f.numRows(), 0, 0);
+    return fillFromImpl(f, 1, (int)f.numRows(), 0, -1, false);
   }
 
-  public FrameV3 fillFromImpl(Frame f, long row_offset, int row_count, int column_offset, int column_count) {
-    if( row_count == 0 ) row_count = 100;                                 // 100 rows by default
-    if( column_count == 0 ) column_count = f.numCols() - column_offset; // full width by default
+  private FrameV3 fillFromImpl(Frame f, long row_offset, int row_count,
+                              int column_offset, int column_count,
+                              boolean expensive) {
+    if( row_count < 0 ) row_count = 100;                                 // 100 rows by default
```

Comment:
This could be an issue where

Reply:
yup, this is breaking REST API contract, but we can fix it on client side (send -1 instead of 0).

Reply:
So do we need to change R client as well?